Error submitting the PR to opam-repository

```
[-] Opening pull request to merge branch release-ocamlformat-0.12 of gh:Julow/opam-repository into ocaml/opam-repository
dune-release: [ERROR] Could not find html_url id in response:
                      HTTP/1.1 100 Continue
                      
                      HTTP/1.1 422 Unprocessable Entity
                      Date: Mon, 04 Nov 2019 16:28:45 GMT
                      Content-Type: application/json;
                      charset=utf-8
                      Content-Length: 235
                      Server: GitHub.com
                      Status: 422 Unprocessable Entity
                      X-RateLimit-Limit: 5000
                      X-RateLimit-Remaining: 4999
                      X-RateLimit-Reset: 1572888525
                      X-OAuth-Scopes: public_repo
                      X-Accepted-OAuth-Scopes: 
                      X-GitHub-Media-Type: github.v3;
                      format=json
                      Access-Control-Expose-Headers: ETag, Link, Location,
                      Retry-After, X-GitHub-OTP, X-RateLimit-Limit,
                      X-RateLimit-Remaining, X-RateLimit-Reset,
                      X-OAuth-Scopes, X-Accepted-OAuth-Scopes,
                      X-Poll-Interval,
                      X-GitHub-Media-Type
                      Access-Control-Allow-Origin:
                      *
                      Strict-Transport-Security: max-age=31536000;
                      includeSubdomains; preload
                      X-Frame-Options: deny
                      X-Content-Type-Options: nosniff
                      X-XSS-Protection: 1; mode=block
                      Referrer-Policy: origin-when-cross-origin,
                      strict-origin-when-cross-origin
                      Content-Security-Policy: default-src
                      'none'
                      X-GitHub-Request-Id:
                      ADF2:FEAB:B5F4D4:DC66C1:5DC051BC
                      
                      {
                        "message": "Validation Failed",
                        "errors": [
                          {
                            "resource": "PullRequest",
                            "field": "head",
                            "code": "invalid"
                          }
                        ],
                        "documentation_url":
                      "https://developer.github.com/v3/pulls/#create-a-pull-request"
                      }
                      .
```

The `base` field of the request is `ocaml-ppx:opam-repository`.
The uri to my fork of `opam-repository` is `gh:Julow/opam-repository`, the github name should be `Julow:opam-repository`.
My `user` field in `~/.config/dune/release.yml` was `ocaml-ppx` but the error persist after changing it.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -11,6 +11,8 @@
 - Separate packages names by spaces in `publish` logs (#171, @hannesm)
 - Fix uncaught exceptions in distrib subcommand and replace them with proper
   error messages (#176, @gpetiot)
+- Use the 'user' field in the configuration before inferring it from repo URI
+  and handles HTTPS URIs (#183, @gpetiot)
 
 ### Removed
 

--- a/bin/opam.ml
+++ b/bin/opam.ml
@@ -116,7 +116,7 @@ let pp_opam_repo fmt opam_repo =
   Format.fprintf fmt "%s/%s" user repo
 
 let open_pr
-    ~dry_run ~changes ~remote_repo ~distrib_user ~branch ~token ~title ~opam_repo ~auto_open ~yes
+    ~dry_run ~changes ~remote_repo ~user ~distrib_user ~branch ~token ~title ~opam_repo ~auto_open ~yes
     pkg =
   Pkg.opam_descr pkg >>= fun (syn, _) ->
   Pkg.opam_homepage pkg >>= fun homepage ->
@@ -134,11 +134,6 @@ let open_pr
       (pp_link "Documentation") doc
       pp_space ()
       changes
-  in
-  let user =
-    match Github.Parse.user_from_remote remote_repo with
-    | Some user -> user
-    | None -> distrib_user
   in
   Prompt.confirm_or_abort ~yes ~question:(fun l -> l "Open PR to %a?" pp_opam_repo opam_repo)
   >>= fun () ->
@@ -173,7 +168,7 @@ let open_pr
       | Ok ()   -> Ok 0
       | Error _ -> msg ()
 
-let submit ~dry_run ~yes ~opam_repo local_repo remote_repo pkgs auto_open =
+let submit ~dry_run ~yes ~opam_repo ~user local_repo remote_repo pkgs auto_open =
   Config.token ~dry_run () >>= fun token ->
   List.fold_left (fun acc pkg ->
       get_pkg_dir pkg
@@ -195,12 +190,20 @@ let submit ~dry_run ~yes ~opam_repo local_repo remote_repo pkgs auto_open =
   let title = strf "[new release] %a (%s)" (pp_list Fmt.string) names version in
   Pkg.publish_msg pkg >>= fun changes ->
   Pkg.distrib_user_and_repo pkg >>= fun (distrib_user, repo) ->
+  let user =
+    match user with
+    | Some user -> user (* from the .yaml configuration file *)
+    | None ->
+        match Github.Parse.user_from_remote remote_repo with
+        | Some user -> user (* trying to infer it from the remote repo URI *)
+        | None -> distrib_user
+  in
   let changes = rewrite_github_refs distrib_user repo changes in
   let msg = strf "%s\n\n%s\n" title changes in
   App_log.status (fun l -> l "Preparing pull request to %a" pp_opam_repo opam_repo);
   Opam.prepare ~dry_run ~msg ~local_repo ~remote_repo ~opam_repo ~version names
   >>= fun branch ->
-  open_pr ~dry_run ~changes ~remote_repo ~distrib_user ~branch ~token ~title ~opam_repo ~auto_open
+  open_pr ~dry_run ~changes ~remote_repo ~user ~distrib_user ~branch ~token ~title ~opam_repo ~auto_open
     ~yes pkg
 
 let field pkgs field = match field with
@@ -274,7 +277,7 @@ let opam () dry_run build_dir local_repo remote_repo opam_repo user keep_v
             | None   -> R.error_msg "Unknown remote repository.")
         >>= fun remote_repo ->
         App_log.status (fun m -> m "Submitting %a" Fmt.(list ~sep:sp Text.Pp.name) pkg_names);
-        submit ~dry_run ~yes ~opam_repo local_repo remote_repo pkgs auto_open
+        submit ~dry_run ~yes ~opam_repo ~user:config.user local_repo remote_repo pkgs auto_open
     | `Field -> field pkgs field_name
   end
   |> Cli.handle_error

--- a/lib/github.ml
+++ b/lib/github.ml
@@ -14,14 +14,17 @@ module D = struct
 end
 
 module Parse = struct
-  let user_from_remote remote_uri =
-    let ssh_uri_regexp =
-      Re.Emacs.compile_pat "git@github\\.com:\\(.+\\)/.+\\(\\.git\\)?"
-    in
-    try
-      let substrings = Re.exec ssh_uri_regexp remote_uri in
-      Some (Re.Group.get substrings 1)
+  let user_from_regexp_opt uri regexp =
+    try Some (Re.(Group.get (exec (Emacs.compile_pat regexp) uri) 1))
     with Not_found -> None
+
+  let user_from_remote uri =
+    match uri with
+    | _ when Bos_setup.String.is_prefix uri ~affix:"git@" ->
+        user_from_regexp_opt uri "git@github\\.com:\\(.+\\)/.+\\(\\.git\\)?"
+    | _ when Bos_setup.String.is_prefix uri ~affix:"https://" ->
+        user_from_regexp_opt uri "https://github\\.com/\\(.+\\)/.+\\(\\.git\\)?"
+    | _ -> None
 
   let archive_upload_url response =
     let open Re in

--- a/lib/github.mli
+++ b/lib/github.mli
@@ -10,9 +10,13 @@ open Bos_setup
 
 module Parse : sig
   val user_from_remote : string -> string option
-  (** [user_from_remote remote_uri] is the username in the github URI [remote_uri]
-      ie [user_from_remote_uri "git@github.com:username/repo.git"] is [Some "username"].
-      Returns [None] if [remote_uri] isn't in the expected format.
+  (** [user_from_remote remote_uri] is the username in the github URI [remote_uri], ie:
+
+      - [user_from_remote_uri "git@github.com:username/repo.git"] returns
+        [Some "username"]
+      - [user_from_remote_uri "https://github.com/username/repo.git"] returns
+        [Some "username"].
+      - Returns [None] if [remote_uri] isn't in the expected format.
   *)
 
   val archive_upload_url : string -> (string, R.msg) Result.result

--- a/tests/test_github.ml
+++ b/tests/test_github.ml
@@ -15,7 +15,7 @@ let user_from_remote () =
   check "git@github.com:user-name-123/repo" (Some "user-name-123");
   check "git@github.com:123/repo" (Some "123");
   check "wrong" None;
-  check "https://github.com/username/repo.git" None
+  check "https://github.com/username/repo.git" (Some "username")
 
 let archive_upload_url () =
   let check response expected =

